### ISSUE TITLE
Buffer module ends up in production build

### ISSUE THREAD

**Issue by davscro** [mzabriskie/axios#901]
**Do you want to request a *feature* or report a *bug*?**

Bug

**What is the current behavior?**

I'm not sure is this bug or excepted behaviour but I believe it's a bug. Running ```yarn build``` to generate production build, **buffer** module ends up in production build!

```
 "scripts": {
    "build": "set NODE_ENV=production rimraf build/dist && webpack -p --config webpack/webpack.config.js && npm run bundle_check",
    "bundle_check": "babel-node bundle_check.js"
  },
```

Could someone explain me why this package ends up in production build. Am I missing something?

<img width="697" alt="screen shot 2017-05-13 at 00 43 18" src="https://cloud.githubusercontent.com/assets/1383250/26035181/7afea984-38c8-11e7-8ac8-83aece9af004.png">

```yarn why buffer```

<img width="538" alt="screen shot 2017-05-13 at 00 45 25" src="https://cloud.githubusercontent.com/assets/1383250/26019467/b8e25862-3775-11e7-8785-5fcfe0d7d81b.png">

```yarn why node-libs-browser```
<img width="444" alt="screen shot 2017-05-13 at 00 45 57" src="https://cloud.githubusercontent.com/assets/1383250/26019479/cea28f14-3775-11e7-9fbe-916ce2976fb1.png">

And webpack is specified only as dev dependency.
I pointed this issue to [WEBPACk](https://github.com/webpack/webpack/issues/4872) but they answered it's connected with axios. Then I searched your repo and found the usage within the **HTTP adapter**

**What is the expected behavior?**
Buffer module should not be in production build.

**Please mention other relevant information such as the browser version, Node.js version, webpack version and Operating System.**

 axios: 0.16.1,
Webpack: 2.5.1
Node: 7.8.0
OS: OSX Sierra

Thanks in advance,
Davor

**Comment by nickuraltsev** [comment]
Fixed via #887.

**Comment by davscro** [comment]
hey @nickuraltsev, #887 is not fixing this issue.
